cmd/engine/main.go

```diff
@@ -331,29 +333,11 @@ func main() { //nolint:gocyclo
 			}
 		}
 
-		// Create a "private" in memory listener for the server that allows us to do some
-		// server setup before officially starting the server on the unix socket listeners
-		// used by actual clients. This enables, for example, clients to wait for cache
-		// mounts to be synchronized before actually connecting and starting to use them.
-		bklog.G(ctx).Debug("creating engine in mem listener")
-		memListener := newInMemListener(server)
-		memClient, err := memListener.NewClient(ctx)
-		if err != nil {
-			return err
-		}
-
-		bklog.G(ctx).Debug("creating engine operator client")
-		daggerClient, stopOperatorSession, err := NewOperatorClient(ctx, memClient)
-		if err != nil {
-			return err
-		}
-		defer stopOperatorSession()
 		bklog.G(ctx).Debug("starting optional cache mount synchronization")
-		err = cacheManager.StartCacheMountSynchronization(ctx, daggerClient)
+		err = cacheManager.StartCacheMountSynchronization(ctx)
 		if err != nil {
-			cancel()
 			bklog.G(ctx).WithError(err).Error("failed to start cache mount synchronization")
-			return err
+			// continue on, doesn't need to be fatal
```

Comment:
This never *should* fail, but in all likelihood there will be some bug lurking somewhere (or even just ephemeral network errors) and I wanted to make sure that the engine can still start even if that happens. Would rather have an empty cache mount than an engine refusing to run.

That being said, this made me realize that we should probably make sure that the cache mount is emptied out if an error happens. It would be bad if the failure happened in the middle of unpacking data to the cache mount and left it in a corrupted state. I *suppose* that users probably need to handle that situation anyways since in theory their own applications could crash and leave these in an invalid state, but there's no reason for us to increase the possibility of that happening. Should be a quick update.

Reply:
Pushed an additional commit with this: https://github.com/dagger/dagger/pull/5050/commits/44f10e4ed5f7b1dae28813d2e94014b369c531ff